internal/providers/amazon/objectstore.go

```diff
@@ -269,41 +263,38 @@ func (s *objectStore) CheckBucket(bucketName string) error {
 func (s *objectStore) ListBuckets() ([]*objectstore.BucketInfo, error) {
 	logger := s.getLogger()
 
-	logger.Info("retrieving bucket list")
-
-	buckets, err := s.objectStore.ListBuckets()
+	logger.Info("retrieving buckets from provider...")
+	s3Buckets, err := s.objectStore.ListBuckets()
 	if err != nil {
-		return nil, err
+		return nil, emperror.Wrap(err, "failed to retrieve buckets")
 	}
 
-	logger.Infof("retrieving managed buckets")
+	logger.Info("retrieving managed buckets...")
+	var managedBuckets []*ObjectStoreBucketModel
 
-	var amazonBuckets []*ObjectStoreBucketModel
-
-	err = s.db.Where(&ObjectStoreBucketModel{OrganizationID: s.org.ID}).Order("name asc").Find(&amazonBuckets).Error
+	err = s.db.Where(&ObjectStoreBucketModel{OrganizationID: s.org.ID}).Order("name asc").Find(&managedBuckets).Error
 	if err != nil {
-		return nil, fmt.Errorf("retrieving managed buckets failed: %s", err.Error())
+		return nil, emperror.Wrap(err, "failed to retrieve managed buckets")
 	}
 
 	var bucketList []*objectstore.BucketInfo
-	for _, bucket := range buckets {
-		// amazonBuckets must be sorted in order to be able to perform binary search on it
-		idx := sort.Search(len(amazonBuckets), func(i int) bool {
-			return strings.Compare(amazonBuckets[i].Name, bucket) >= 0
+	for _, bucket := range s3Buckets {
+		// managedBuckets must be sorted in order to be able to perform binary search on it
+		idx := sort.Search(len(managedBuckets), func(i int) bool {
+			return strings.Compare(managedBuckets[i].Name, bucket) >= 0
 		})
 
 		bucketInfo := &objectstore.BucketInfo{Name: bucket, Managed: false}
-		if idx < len(amazonBuckets) && strings.Compare(amazonBuckets[idx].Name, bucket) == 0 {
+		if idx < len(managedBuckets) && strings.Compare(managedBuckets[idx].Name, bucket) == 0 {
 			bucketInfo.Managed = true
 		}
 
-		region, err := s.objectStore.GetRegion(bucket)
-		if err != nil {
-			return nil, err
+		if region, err := s.objectStore.GetRegion(bucket); err == nil {
+			bucketInfo.Location = region
+			bucketList = append(bucketList, bucketInfo)
+		} else {
+			logger.WithField("bucket", bucket).WithError(err).Warn("skipping bucket from the list")
```

Comment:
I think we may need to differentiate between error codes here: if the bucket is not found, it can be skipped, but this API call can fail for other reasons as well, like missing permissions.
In those cases we should still return with an error.
see [docs](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/handling-errors.html) about error handling

Reply:
@martonsereg  thanks for spotted this out; fixed